src/state/slices/channel_slice.ts

```diff
@@ -129,11 +119,31 @@ export const addChannelDerivedStateSubscribers = (
   );
 };
 
-// TODO: Implement serialization
-export const serializeChannelSlice = (_slice: Partial<ChannelSliceSerializableState>): SerializedStoreData => ({});
+export const serializeChannelSlice = (slice: Partial<ChannelSliceSerializableState>): SerializedStoreData => {
+  const channelSettings = slice.channelSettings;
+  const ret: SerializedStoreData = {};
+  channelSettings?.forEach((setting, index) => {
+    ret[`c${index}`] = encodeChannelSetting(setting);
+  });
+  return ret;
+};
 
 export const selectChannelSliceSerializationDeps = (slice: ChannelSlice): ChannelSliceSerializableState => ({
   channelSettings: slice.channelSettings,
 });
 
-export const loadChannelSliceFromParams = (_slice: ChannelSlice, _params: URLSearchParams): void => {};
+export const loadChannelSliceFromParams = (slice: ChannelSlice, params: URLSearchParams): void => {
+  // Find all params matching channel format
+  const channelSettings: Partial<ChannelSetting>[] = [];
+  for (const [key, value] of params.entries()) {
+    if (isChannelKey(key) && value) {
+      const channelIndex = parseInt(key.slice(1), 10);
+      if (Number.isNaN(channelIndex)) {
+        channelSettings.push({});
```

Comment:
Looks like `channelSettings` is only ever referenced in order to push empty objects into it. Still needed?